fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/ComposeTopSitesRobot.kt

```diff
@@ -0,0 +1,170 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui.robots
+
+import android.widget.EditText
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.filter
+import androidx.compose.ui.test.hasAnyChild
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.hasText
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.longClick
+import androidx.compose.ui.test.onAllNodesWithTag
+import androidx.compose.ui.test.onFirst
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.compose.ui.test.performTouchInput
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import org.hamcrest.CoreMatchers
+import org.hamcrest.Matchers
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.HomeActivityComposeTestRule
+import org.mozilla.fenix.helpers.MatcherHelper.itemContainingText
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.home.topsites.TopSitesTestTag
+
+/**
+ * Implementation of Robot Pattern for the Compose Top Sites.
+ */
+class ComposeTopSitesRobot(private val composeTestRule: HomeActivityComposeTestRule) {
+
+    fun verifyExistingTopSitesList() =
+        composeTestRule.onNodeWithTag(TopSitesTestTag.topSites).assertExists()
+
+    @OptIn(ExperimentalTestApi::class)
+    fun verifyExistingTopSiteItem(vararg titles: String) {
+        titles.forEach { title ->
+            itemContainingText(title).waitForExists(waitingTime)
+            composeTestRule.waitUntilAtLeastOneExists(hasText(title), waitingTime)
+            composeTestRule.topSiteItem(title).assertExists()
+        }
+    }
+
+    fun verifyNotExistingTopSiteItem(vararg titles: String) {
+        titles.forEach { title ->
+            itemContainingText(title).waitForExists(waitingTime)
+            composeTestRule.topSiteItem(title).assertDoesNotExist()
+        }
+    }
+
+    fun verifyTopSiteContextMenuItems() {
+        verifyTopSiteContextMenuOpenInPrivateTabButton()
+        verifyTopSiteContextMenuRemoveButton()
```

Comment:
Could you please add the rename option as well ? (`verifyTopSiteContextMenuRenameButton`)